### PR TITLE
State without trie

### PR DESCRIPTION
**Pull request by mikhailOK**
TODO:
- [ ] forks management
- [ ] correct propagation of block info in state sync and some test code

### CODE REVIEW

File: core/store/src/runtime_state/state_changes.rs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    /// Add kv changes to StoreUpdate transaction
+    fn key_value_changes_into(
+        &self,
+        trie: Arc<Trie>,
+        block_hash: CryptoHash,
+        store_update: &mut StoreUpdate,
+    ) -> Result<(), StorageError>;
+
+    /// Return a StoreUpdate transaction with insertions and deletions (but not kv changes)
+    fn into2(&self, trie: Arc<Trie>) -> Result<(StoreUpdate, StateRoot), StorageError> {
```
Comment (Kouprin) [inline core/store/src/runtime_state/state_changes.rs]: into2??

File: near/src/runtime.rs [hunk 2]
```diff
@@ -1053,8 +1065,14 @@ impl RuntimeAdapter for NightshadeRuntime {
         let trie_changes = Trie::combine_state_parts(&state_root, &parts)
             .expect("combine_state_parts is guaranteed to succeed when each part is valid");
         // TODO clean old states
-        let trie = self.trie.clone();
-        let (store_update, _) = trie_changes.into(trie).expect("TrieChanges::into never fails");
+        let mut store_update = StoreUpdate::new_with_trie(self.trie.clone());
+        // TODO pass correct block hash
+        let flat_db_changes =
+            FlatDBChanges { shard_id, block_hash: Default::default(), block_height };
```
Comment (Kouprin) [inline near/src/runtime.rs]: is it a `sync_hash` which can be taken in `set_state_finalize()`?

File: near/src/runtime.rs [hunk 3]
```diff
@@ -1053,8 +1065,14 @@ impl RuntimeAdapter for NightshadeRuntime {
         let trie_changes = Trie::combine_state_parts(&state_root, &parts)
             .expect("combine_state_parts is guaranteed to succeed when each part is valid");
         // TODO clean old states
```
Comment (Kouprin) [inline near/src/runtime.rs]: does `clean old states` now relates to erasing data from ColFlatState, not trie?

File: runtime/runtime/src/ext.rs [hunk 4]
```diff
@@ -361,10 +361,11 @@ impl<'a> External for RuntimeExt<'a> {
     }
 
     fn get_touched_nodes_count(&self) -> u64 {
-        self.trie_update.trie.counter.get()
+        0
```
Comment (Kouprin) [inline runtime/runtime/src/ext.rs]: remove?

**Comment by ilblackdragon** [comment]
Closing this until we reopen this item.

**Pull request closed without merging**